oskari-geojson/src/main/java/org/oskari/geojson/GeoJSONReader2.java

```diff
@@ -0,0 +1,311 @@
+package org.oskari.geojson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.geometry.jts.JTS;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.geometry.MismatchedDimensionException;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryCollection;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.MultiLineString;
+import com.vividsolutions.jts.geom.MultiPoint;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+
+/**
+ * Convert Map<String, Object> (that follow GeoJSON spec) to
+ *  - GeoTools SimpleFeatureCollection
+ *  - JTS Geometries
+ * For invalid input we throw IllegalArgumentExceptions
+ */
+public class GeoJSONReader2 {
+
+    private static final GeometryFactory GF = new GeometryFactory();
+
+    /**
+     * Replaces geometries that are currently represented as Map<String, Object>
+     * with JTS Geometries in the map
+     * Replaces both the default geometries ('geometry' key) as well as entries
+     * in the properties map that were eligible geometries
+     */
+    public static void replaceFeatureCollectionsMapsWithGeometries(Map<String, Object> featureCollection) {
+        String type = GeoJSONUtil.getString(featureCollection, GeoJSON.TYPE);
+        if (!GeoJSON.FEATURE_COLLECTION.equals(type)) {
+            if (GeoJSON.FEATURE.equals(type)) {
+                replaceFeaturesMapWithGeometries(featureCollection);
```

Comment:
If not the above this could have a comment saying:

    // expected collection but got a feature -> redirecting to proper method.